test.lua

```diff
@@ -781,6 +781,68 @@ do
     end
 end
 
+---- REPL tests ----
+print("Running tests for REPL completion...")
+local wrapRepl = function()
+    local replComplete
+    local replSend = coroutine.wrap(function()
+        local output = {}
+        fennel.repl({
+            readChunk = function()
+                local chunk = coroutine.yield(output)
+                output = {}
+                return chunk and chunk .. '\n' or nil
+            end,
+            onValues = function(xs)
+                table.insert(output, xs)
+            end,
+            registerCompleter = function(completer)
+                replComplete = function(input)
+                    local res = completer(input)
+                    table.sort(res)
+                    return res
+                end
+            end,
+            pp = function(x) return x end,
+        })
+    end) replSend()
+    return replSend, replComplete
+end
+
+-- Skip REPL tests in non-JIT Lua 5.1 only to avoid engine coroutine bug
+if _VERSION ~= 'Lua 5.1' or type(jit) == 'table' then
+    local send, comp = wrapRepl()
+    local v = {
```

Comment:
I don't understand the purpose of the `v` table here. Wouldn't it be simpler just to make `wrapRepl` return functions which are usable as-is?

Reply:
Initially I wanted `wrapRepl` generalized - per my comment above, the `v` table is mostly just a workaround for `deep_equal` being broken, and later when it's fixed `wrapRepl` could be left the same and the `v` table just removed. I'm comfortable removing it now and just updating `wrapRepl` though.

Reply:
Removing the `v` table!